src/Sulu/Bundle/PageBundle/Resources/js/containers/Form/fields/PageSettingsShadowLocaleSelect.js

```diff
@@ -21,25 +22,31 @@ export default class PageSettingsShadowLocaleSelect extends React.Component<Fiel
             throw new Error('The "contentLocales" should be an array!');
         }
 
+        const filteredContentLocales = contentLocales.filter(
+            (contentLocale) => locale && contentLocale !== locale.get()
+        );
+
         return (
             <SingleSelect disabled={!!disabled} onChange={this.handleChange} value={value}>
-                {contentLocales
-                    .filter((contentLocale) => locale && contentLocale !== locale.get())
-                    .map((contentLocale) => {
-                        if (typeof contentLocale !== 'string') {
-                            throw new Error('All entries in the "contentLocales" array must be strings!');
-                        }
+                {filteredContentLocales.length > 0 && filteredContentLocales.map((contentLocale) => {
+                    if (typeof contentLocale !== 'string') {
+                        throw new Error('All entries in the "contentLocales" array must be strings!');
+                    }
 
-                        return (
-                            <SingleSelect.Option
-                                key={contentLocale}
-                                value={contentLocale}
-                            >
-                                {contentLocale}
-                            </SingleSelect.Option>
-                        );
-                    })
-                }
+                    return (
+                        <SingleSelect.Option
+                            key={contentLocale}
+                            value={contentLocale}
+                        >
+                            {contentLocale}
+                        </SingleSelect.Option>
+                    );
+                })}
+                {filteredContentLocales.length === 0 && (
```

Comment:
Shouldn't that kind of code be in the `Select` component? I think that we might have the same situation in multiple places.

We could introduce a new prop called `noOptionsText` or something like that, but I think I would even prefer a generic text here (the label is already "Shadow Locale", then a text like "No options available" would cut it for me as well).